pkg/daemon/ceph/multus/mover.go

```diff
@@ -0,0 +1,437 @@
+			logger.Infof("un-moving interface for holder pod %q with info: %+v", holderPodName, info)
+
+			if err := cleanUpMovedInterface(holderPodName, movedInterfaceCache); err != nil {
+				logger.Errorf("failed to clean up interface for holder pod %q during termination; "+
+					"continuing despite failure", holderPodName)
+			}
+		}
+
+		// clean up any remaining interfaces that might not have been cleaned up for some reason
+		cleanupAllInterfacesWithPrefix(MovedNetworkInterfacePrefix)
+	}()
+
+	// NOTE: this blocks until the context is canceled
+	err = mgr.Start(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to start %q controller", MoverControllerName)
+	}
+
+	return nil
+}
+
+type HolderPodReconciler struct {
+	client.Client
+	NetworkMutex *sync.Mutex // use during network move/cleanup operations to avoid races
+	MovedCache   *MovedInterfaceCache
+}
+
+func (r *HolderPodReconciler) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	// wrap the reconcile so we can log errors with Rook's logger
+	res, err := r.reconcile(ctx, req)
+	if err != nil {
+		logger.Errorf("reconcile failed for pod %q. %s", req.Name, err.Error())
+		if !res.IsZero() {
+			// if there is an error and the result is nonzero, pass the result to controller-runtime
+			// with a nil-ed error so that we can delay re-reconciles a bit even in the error case
+			err = nil
+		}
+	}
+	return res, err
+}
+
+func (r *HolderPodReconciler) reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	if ctx.Err() != nil {
+		logger.Infof("not reconciling holder pod %q since reconciliation is canceled", req.Name)
+		return controllerruntime.Result{}, nil // canceled, nothing more to do
+	}
+
+	pod := &corev1.Pod{}
+	err := r.Get(ctx, req.NamespacedName, pod)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			//
+			// clean up interface of deleted pod
+			//
+			logger.Infof("holder pod %q was deleted; ensuring its interface is cleaned up", req.Name)
+			return r.cleanUpMovedInterface(req.Name)
+
+		}
+		return controllerruntime.Result{},
+			errors.Wrapf(err, "failed to get kubernetes pod resource for holder pod %q", req.Name)
+	}
+	if pod.ObjectMeta.DeletionTimestamp != nil {
+		//
+		// clean up interface of pod marked for deletion
+		//
+		logger.Infof("holder pod %q is marked for deletion; ensuring its interface is cleaned up", req.Name)
+		return r.cleanUpMovedInterface(pod.Name)
+	}
+
+	//
+	// move interface
+	//
+	logger.Infof("ensuring interface for holder pod %q is moved to the host network namespace", pod.Name)
+	return r.ensureMovedInterface(ctx, pod)
+}
+
+// should be idempotent and good at making sure interfaces exist, so don't skip any steps if the
+// holder pod is already in the moved interface cache
+func (r *HolderPodReconciler) ensureMovedInterface(ctx context.Context, holderPod *corev1.Pod) (controllerruntime.Result, error) {
```

Comment:
Should be called copied instead of moved? It's the internal implementation so it's probably useful.